Index: ../release/release.sh
===================================================================
--- ../release/release.sh	(revision 259943)
+++ ../release/release.sh	(working copy)
@@ -127,9 +127,15 @@
 	ARCH_FLAGS=
 fi
 CHROOT_MAKEENV="MAKEOBJDIRPREFIX=${CHROOTDIR}/tmp/obj"
-CHROOT_WMAKEFLAGS="${MAKE_FLAGS} ${WORLD_FLAGS} ${CONF_FILES}"
-CHROOT_IMAKEFLAGS="${CONF_FILES}"
-CHROOT_DMAKEFLAGS="${CONF_FILES}"
+if [ "$(sysctl -n hw.machine)" = "amd64" ] && [ "${TARGET}" = "i386" ] \
+	&& [ "${TARGET_ARCH}" = "i386" ]; then
+	CHROOT_EXTRAFLAGS="${ARCH_FLAGS}"
+else
+	CHROOT_EXTRAFLAGS="TARGET=amd64 TARGET_ARCH=amd64"
+fi
+CHROOT_WMAKEFLAGS="${MAKE_FLAGS} ${WORLD_FLAGS} ${CONF_FILES} ${CHROOT_EXTRAFLAGS}"
+CHROOT_IMAKEFLAGS="${CONF_FILES} ${CHROOT_EXTRAFLAGS}"
+CHROOT_DMAKEFLAGS="${CONF_FILES} ${CHROOT_EXTRAFLAGS}"
 RELEASE_WMAKEFLAGS="${MAKE_FLAGS} ${WORLD_FLAGS} ${ARCH_FLAGS} ${CONF_FILES}"
 RELEASE_KMAKEFLAGS="${MAKE_FLAGS} ${KERNEL_FLAGS} KERNCONF=\"${KERNEL}\" ${ARCH_FLAGS} ${CONF_FILES}"
 RELEASE_RMAKEFLAGS="${ARCH_FLAGS} KERNCONF=\"${KERNEL}\" ${CONF_FILES} \
@@ -163,12 +169,6 @@
 	svn co ${SVNROOT}/${PORTBRANCH} ${CHROOTDIR}/usr/ports
 fi
 
-cd ${CHROOTDIR}/usr/src
-env ${CHROOT_MAKEENV} make ${CHROOT_WMAKEFLAGS} buildworld
-env ${CHROOT_MAKEENV} make ${CHROOT_IMAKEFLAGS} installworld \
-	DESTDIR=${CHROOTDIR}
-env ${CHROOT_MAKEENV} make ${CHROOT_DMAKEFLAGS} distribution \
-	DESTDIR=${CHROOTDIR}
 mount -t devfs devfs ${CHROOTDIR}/dev
 cp /etc/resolv.conf ${CHROOTDIR}/etc/resolv.conf
 trap "umount ${CHROOTDIR}/dev" EXIT # Clean up devfs mount on exit
